backend/dataall/modules/dataset_sharing/services/share_managers/s3_access_point_share_manager.py

```diff
@@ -140,34 +142,53 @@ def grant_target_role_access_policy(self):
         logger.info(
             f'Grant target role {self.target_requester_IAMRoleName} access policy'
         )
+        key_alias = f"alias/{self.dataset.KmsAlias}"
+        kms_client = KmsClient(self.dataset_account_id, self.source_environment.region)
+        kms_key_id = kms_client.get_key_id(key_alias)
+
         existing_policy = IAM.get_role_policy(
             self.target_account_id,
             self.target_requester_IAMRoleName,
-            "targetDatasetAccessControlPolicy",
+            IAM_ACCESS_POINT_ROLE_POLICY,
         )
         if existing_policy:  # type dict
-            if self.bucket_name not in ",".join(existing_policy["Statement"][0]["Resource"]):
-                logger.info(
-                    f'targetDatasetAccessControlPolicy exists for IAM role {self.target_requester_IAMRoleName}, '
-                    f'but S3 Access point {self.access_point_name} is not included, updating...'
-                )
-                target_resources = [
+            s3_target_resources = [
                     f"arn:aws:s3:::{self.bucket_name}",
```

Comment:
identation formatting seems to be wrong

Reply:
It looks right to me in my editor. Could you please describe exactly what looks wrong?